timestampvm/vm.go

```diff
@@ -230,26 +294,47 @@ func (vm *VM) ParseBlock(bytes []byte) (snowman.Block, error) {
 // - the block's parent is [parentID]
 // - the block's data is [data]
 // - the block's timestamp is [timestamp]
-// The block is persisted in storage
-func (vm *VM) NewBlock(parentID ids.ID, height uint64, data [dataLen]byte, timestamp time.Time) (*Block, error) {
-	// Create our new block
-	block := &Block{
-		Block: core.NewBlock(parentID, height, timestamp.Unix()),
-		Data:  data,
-	}
+func (vm *VM) NewBlock(parentID ids.ID, height uint64, data [dataLen]byte, timestamp time.Time) (Block, error) {
+	block := newTimeBlock(parentID, height, data, timestamp)
```

Comment:
it feels kinda weird that we would have a `NewBlock` and a `newTimeBlock` - seeing as there is only one block type.